public/app/features/dashboard-scene/scene/DashboardScene.tsx

```diff
@@ -328,10 +328,14 @@ export class DashboardScene extends SceneObjectBase<DashboardSceneState> impleme
     // No need to listen to changes anymore
     this._changeTracker.stopTrackingChanges();
 
+    const initialTimeZone = this._initialState?.$timeRange?.state.timeZone;
+    const currentTimeZone = this.state.$timeRange!.state.timeZone ?? initialTimeZone;
+
     // We are updating url and removing editview and editPanel.
     // The initial url may be including edit view, edit panel or inspect query params if the user pasted the url,
     // hence we need to cleanup those query params to get back to the dashboard view. Otherwise url sync can trigger overlays.
     const url = locationUtil.getUrlForPartial(this._initialUrlState!, {
+      timezone: restoreInitialState ? initialTimeZone : currentTimeZone,
```

Comment:
I do not understand this, why is this needed?

Reply:
something wrong with this function why are we updating the URL with initial state even when restoreInitialState is false?? 

Should not line 342 only be executed when restoreInitialState is true?

Reply:
@torkelo I think that makes sense, I questioned this as well but gave up the thought thinking that we want to always restore mainly because of timerange and variables. Indeed the timezone thing is a bug because we also add that to the URL, but otherwise I was thinking that we want to always revert variable changes and timerange changes to whatever was before edit mode

Reply:
@mdvictor yea, but since _initialUrlState is only set when going into edit mode (and never updated on saved), it makes 0 sense to use it when leaving edit mode (unless your discarding changes), but even then it could be a bug as those changes could have been done after you saved, so you don't want to go back to the URL state of before your go into edit mode (ie before you made a bunch of changes and saved), that could be very buggy. 

I think if we update _initialUrlState when we save this logic might work

Reply:
but then you should need no special handling of timezone

Reply:
in which case this also diverges and may be unexpected? when we have no changes in the model it will maintain variable and timerange changes, but if we do change the model, e.g.: add a panel, and then exit and discard without saving, which turns the `restoreInitialState ` flag to true, then we also revert to whatever timerange and variable values were in the initial url state... so that may be a bit confusing?